src/components/react/chat.tsx

```diff
@@ -15,40 +17,71 @@ const Chat: React.FC<ChatProps> = (props) => {
   );
 };
 
-const InnerChat: React.FC<ChatProps> = ({ id, type }) => {
-  const messagesRes = useQuery({
-    queryKey: ["messages", type, id],
-    queryFn: async () => {
+const InnerChat: React.FC<ChatProps> = ({ id, type, showRefs = false }) => {
+  const messagesRes = useQuery(
+    ["messages", type, id], // queryKey
+    async () => {
+      // queryFn
       const endpoint = id
         ? `/api/assistant/chat/${id}?type=${type}`
-        : `/api/assistant/chat`;
+        : `/api/assistant/chat/all`;
 
       const response = await fetch(endpoint);
-      const data = (await response.json()) as Promise<{
-        messages: MessageSelect[];
-      }>;
+      const data = await response.json(); // Log the server response
 
-      return data;
+      return data as {
+        messages: MessageSelect[];
+        referencedFiles: { id: string }[];
+      };
     },
-  });
+    {
+      // options
+      refetchOnReconnect: false,
+      refetchOnWindowFocus: false,
+      onSuccess: () => {},
+    }
+  );
 
   const sendMessageMutation = useMutation({
     mutationKey: [type, id, "send-mutation"],
     mutationFn: async (message: string) => {
       const endpoint = id
         ? `/api/assistant/chat/${id}?type=${type}&message=${message}`
-        : `/api/assistant/chat?type=${type}&message=${message}`;
+        : `/api/assistant/chat/all?type=${type}&message=${message}`;
 
       const response = await fetch(endpoint);
       const data = (await response.json()) as Promise<{
         messages: MessageSelect[];
+        referencedFiles: FileReference[];
       }>;
 
       return data;
     },
     onMutate: () => scrollToBottom(),
     onSettled: () =>
       queryClient.invalidateQueries({ queryKey: ["messages", type, id] }),
+    onSuccess: () => {
+      console.log(sendMessageMutation.data?.referencedFiles);
+    },
+  });
+
+  const deleteConversationMutation = useMutation({
+    mutationKey: [id, "delete-conversation-mutation"],
+    mutationFn: async (id: string) => {
+      const endpoint = `/api/conversations/delete`;
```

Comment:
This should probably be DELETE /api/conversations/[id], no need to send body or specify /delete